packages/flutter/lib/src/rendering/object.dart

```diff
@@ -295,17 +323,23 @@ class PaintingContext extends ClipContext {
   /// See also:
   ///
   ///  * [addLayer], for pushing a leaf layer whose canvas is not used.
-  void pushLayer(Layer childLayer, PaintingContextCallback painter, Offset offset, { Rect childPaintBounds }) {
+  void pushLayer(ContainerLayer childLayer, PaintingContextCallback painter, Offset offset, { Rect childPaintBounds }) {
```

Comment:
I believe this method was always supposed to require that childLayer is a ContainerLayer but implicit casts in Dart hid that requirement.